clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -17468,19 +17468,19 @@ Value *EmitAMDGPUImplicitArgPtr(CodeGenFunction &CGF) {
 /// Emit code based on Code Object ABI version.
 /// COV_4    : Emit code to use dispatch ptr
 /// COV_5    : Emit code to use implicitarg ptr
-/// COV_NONE : Emit code to load a global variable "llvm.amdgcn.abi.version"
+/// COV_NONE : Emit code to load a global variable "__oclc_ABI_version"
```

Comment:
I have encountered the need for `__oclc_ABI_version` to be defined while I was fixing the new driver downstream, so I think this should be done at least in the downstream version. There's probably no reason to split the name at this point when they describe the same thing.

Long term, I don't think we should be using things related to the ROCm Device Libraries since they're an external thing. The best solution for this stuff is to either handle it in the backend or just default to some value. I'm going to assume that there will be a grace period where we support COV4, but once we've decided that support isn't needed we can probably just get rid of this code entirely and just expect COV5 or something. Hopefully we don't end up needing more things tied to this logic because I really don't like it.

Reply:
`oclc_ABI_version` already exist in the upstream. And, it contains the exact same information as `llvm.amdgcn.abi.version` was supposed to store. So, I removed the latter with the former.

This patch is ensuring that DeviceRTL is indeed ABI agnostic.

I agree with your suggestion that it would be great to not deal with these things in the driver.